src/fs.register.ts

```diff
@@ -66,62 +66,60 @@ export function eaiAgainBuilder(timeout: (attempt: number) => number): BuildMidd
 /**
  * When a new AWS file system is created copy across the middleware, but only if the middleware does not exist
  *
- * @param fsClient Filesystem to setup
+ * @param fileSystem to setup
  */
-export function setupS3FileSystem(fsClient: FsAwsS3V3): void {
-  addMiddlewareToS3Client(fsClient.client);
-
-  if (fsClient.credentials == null) return;
-  const oldFind = fsClient.credentials.find.bind(fsClient.credentials);
-  // When file systems are looked up ensure they are registered into `fsa`
-  fsClient.credentials.find = async (path: string): Promise<FsAwsS3 | null> => {
-    const accountConfig = await fsClient.credentials.findCredentials(path);
-    if (accountConfig == null) return null;
-
-    const fileSystem = await oldFind(path);
-    if (fileSystem == null) return null;
-
-    logger.debug({ prefix: path, roleArn: accountConfig.roleArn }, 'FileSystem:Register');
-    fsa.register(accountConfig.prefix, fileSystem);
-    if (fileSystem.s3 != null && 'client' in fileSystem.s3) {
-      addMiddlewareToS3Client((fileSystem.s3 as S3LikeV3).client);
-    }
-    return fileSystem;
+export function setupS3FileSystem(fileSystem: FsAwsS3): FsAwsS3 {
+  const credentials = new AwsS3CredentialProvider();
+  credentials.onFileSystemFound = (acc: AwsCredentialConfig, fs?: FsAwsS3, location?: URL): void => {
```

Comment:
Could this re-introduce [the issue where the incorrect credentials are used for a bucket](https://toitutewhenua.atlassian.net/browse/TDE-1268)?

Reply:
I will resolve this comment as the test does pass.